src/main/java/org/spongepowered/common/data/util/DataProcessorDelegate.java

```diff
@@ -83,6 +83,22 @@ public boolean supports(DataHolder dataHolder) {
 
     @Override
     public boolean supports(EntityType entityType) {
+        final boolean callingFromMinecraftThread = ServerUtils.isCallingFromMainThread();
+
+        for (Tuple<DataProcessor<M, I>, Timing> tuple : this.processors) {
+            if (callingFromMinecraftThread) {
```

Comment:
I would write it in a try like so:
```java
try (Timing timing = callingFromMainecraftThread ? tuple.getSecond().startTiming() : null) {
  if (tuple.getFirst().supports(entityType)) {
    return true;
  }
}
return false;

Reply:
This way the timing is auto-closing, in the event the processor royally breaks (there's lots of try with resources in sponge now due to their fancy handling).